File: backend/src/main/java/com/pk/cruncher/entity/Supply.java
package com.pk.cruncher.entity;


import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.OffsetDateTime;
import java.util.List;

@Data
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
@Entity
public class Supply extends BaseEntity {

    @ManyToOne
    @JoinColumn(name = "supplier_id")
    private Supplier supplier;

    private OffsetDateTime suppliedAt;

    @Column(length = 4096)
    private String comment;

    @OneToMany(mappedBy = "supply", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<ItemSupply> itemSupplies;
}

File: src/app/model/supply.ts
import {ItemSupply} from './item-supply';

export interface Supply {
  id: string;
  createdAt: Date;
  updatedAt: Date;
  supplierId: string;
  supplierName: string;
  suppliedAt: Date;
  comment: string;
  itemSupplies: ItemSupply[];
}
